app/controllers/concerns/saml_idp_auth_concern.rb

```diff
@@ -22,12 +22,6 @@ def validate_service_provider_and_authn_context
     render nothing: true, status: :unauthorized
   end
 
-  # stores original SAMLRequest in session to continue SAML Authn flow
-  def store_saml_request_attributes_in_session
```

Comment:
note that this was removed *on purpose* - we are no longer keeping track of the `saml_request_url` session var and are instead using `user_return_to` to send users back to the service provider.

but all session data is deleted during the `auth` request, which is what this spec is for (see `delete_branded_experience` method). That is *totally fine* because the `auth` method calls `render_template_for(saml_response, saml_request.response_url, 'SAMLResponse')` ... and `saml_request.response_url` is what sets the form action for the rendered form (which is submitted with javascript).

In short: we don't care about this session var